File: 739.daily-temperatures/solution.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        answer = [0] * len(temperatures)

        for i, temp in enumerate(temperatures):
            while(stack) and (temperatures[stack[-1]] < temp):
                idx = stack.pop()
                answer[idx] = i - idx

            stack.append(i)
        return answer

File: 739.daily-temperatures/solution.kt
import java.util.Stack
class Solution {
    fun dailyTemperatures(temperatures: IntArray): IntArray {
        val stack = Stack<Int>()
        var answer = IntArray(temperatures.size)

        for ((i, temp) in temperatures.withIndex()) {
            while (!stack.empty() && (temperatures[stack.peek()] < temp)) {
                val idx = stack.pop()
                answer[idx] = i - idx
            }
            stack.push(i)
        }
        return answer
    }
}
